src/index.ts

```diff
@@ -291,12 +344,7 @@ export function setGCPResidency(value: boolean | null = null) {
  * @returns {number} a request timeout duration in milliseconds.
  */
 export function requestTimeout(): number {
-  // Detecting the residency can be resource-intensive. Let's cache the result.
-  if (gcpResidencyCache === null) {
-    gcpResidencyCache = detectGCPResidency();
-  }
-
-  return gcpResidencyCache ? 0 : 3000;
+  return getGCPResidency() ? 0 : 3000;
```

Comment:
Curious what timeout it is and why exactly 0?

Reply:
The timeout is used for availability checking; 0 (indefinite) for GCE and up to 3 secs for other platforms.